### PR TITLE
Rapid Wiring Device(RWD)

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request

Just a big roll of cable(7 stacks worth of cable) you carry around it has the following features
1. Dual wield it and just walk across plating to lay new cable. Make sure you wear insulated gloves
2. If you just have it in one of your hands it can pick up any stray cable pieces lying around
3. Right click on it to dispense a custom amount of cable
4. Refill with cable
5. Alt click to change cable layer
6. Available in autolathe & engineering protolathe round start

https://user-images.githubusercontent.com/110812394/213224959-f393e849-2e0a-4314-ab0e-c10924cf8fce.mp4

## Why It's Good For The Game
1. Clean up cable pieces bitten by rats and lay fresh cable in maintenance very quickly.
2. For big construction projects. Carrying one big roll of 7 stacks of cable is much more storage efficient than carrying 7 separate stacks individually.
3. For wiring Solars very quickly

Also, I'm pretty sure something like this exists in real life so why not. It's just the rapid pipe cleaner but for cables.

## Changelog
:cl:
add: Rapid wiring device
imageadd: Sprite for RWD
/:cl:

### CODE REVIEW

**Comment by BlueMemesauce** [comment]
This used to exist before being removed in #44265, dont know why it took so long to finally get it back.

**Comment by SyncIt21** [comment]
> This used to exist before being removed in #44265, dont know why it took so long to finally get it back.

Damn why was it removed? Well that's in the past now

**Comment by BlueMemesauce** [comment]
The wiring system was updated in that PR, so I think the author just didn't feel like updating the cable layer to the new system.

**Comment by Pickle-Coding** [comment]
How would you change its layer?

**Comment by SyncIt21** [comment]
> How would you change its layer?

You can't sadly. Changing layer would mean changing its colour so i would need to create a whole other sprite to update the device.

**Comment by SyncIt21** [comment]
> How would you change its layer?

You know it is something i want to see. I would need to create a total of 16 new sprites but they are pretty small anyway so its doable. I'll draft this till i implement it

**Comment by Wallemations** [comment]
> > How would you change its layer?
> 
> 
> 
> You know it is something i want to see. I would need to create a total of 16 new sprites but they are pretty small anyway so its do able. I'll draft this till i implement it

I'm pretty sure cables don't have a layer function. If you're going to implement something like that (though I don't know why you would), it would be best to leave it for a different PR imo

**Comment by SyncIt21** [comment]
It does have a layer function.  Layer 2 is yellow(default),  layer 1 is red and layer 3 is blue( I think im correct) just click on the cable and it will show you the raidal menu for it. 
It's pretty simple to implement actually I have no problem with it

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by SyncIt21** [comment]
Something is wrong. I downloaded the latest dmi files from the master & updated them with my sprites and commited but github is still complaining about a merge conflict.
How do i fix this?

**Comment by SyncIt21** [comment]
> How would you change its layer?

You alt click

**Comment by SyncIt21** [comment]
> Also, shouldn't this be a subtype of `/obj/item/rcl`? Or something?

It would be a waste because most of the procs would be overrided cause i can't use its code(because it deals with pipe cleaner coils and not cable coil)

**Comment by SyncIt21** [comment]
Ok i think I'm done. The in hand icon doesn't update when switching layers and i cant figure out why, If you drop & pick it back up it updates correctly but that's just a slight inconvenience not a bug. 
Apart from that everything is tested

File: code/game/objects/items/RWD.dm [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+	desc = "A device used to rapidly lay cable & pick up stray cable pieces laying around."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rcl-0"
+	inhand_icon_state = "rcl-0"
+	opacity = FALSE
+	force = 5 //Plastic is soft
+	throwforce = 5
+	throw_speed = 1
+	throw_range = 7
+	w_class = WEIGHT_CLASS_NORMAL
```
Comment (Fikou) [inline code/game/objects/items/RWD.dm:12]: should be bulky

Comment (BlueMemesauce) [reply code/game/objects/items/RWD.dm:12]: I dont think it needs to be bulky its essentially the same as having a box in your backpack that you fill with cable coils

Comment (SyncIt21) [reply code/game/objects/items/RWD.dm:12]: Code copied from rlc so makes sense it should be the same weight class as that

**Comment by SyncIt21** [comment]
Merge/Review anyone? I have no more pending work on this

**Pull request merged**